applications/nrf_desktop/doc/ble_discovery.rst

```diff
@@ -1,6 +1,71 @@
-.. _ble_discovery:
+.. _nrf_desktop_ble_discovery:
 
-BLE discovery module
-####################
+Bluetooth LE discovery module
+#############################
 
-Use the ``ble_discovery`` module to discover GATT services and read GATT characteristics.
+Use the Bluetooth LE discovery module to discover GATT Services and read GATT Attribute Values from an nRF Desktop peripheral.
+The module is mandatory for the nRF Desktop central.
+
+Module Events
+*************
+
++------------------------------+----------------------------------+-------------------+----------------------------------+--------------------------------+
+| Source Module                | Input Event                      | This Module       | Output Event                     | Sink Module                    |
++==============================+==================================+===================+==================================+================================+
+| :ref:`nrf_desktop_ble_state` | ``ble_peer_event``               | ``ble_discovery`` |                                  |                                |
++------------------------------+----------------------------------+                   |                                  |                                |
+| ``ble_discovery``            | ``ble_discovery_complete_event`` |                   |                                  |                                |
++------------------------------+----------------------------------+                   +----------------------------------+--------------------------------+
+|                              |                                  |                   | ``ble_discovery_complete_event`` | :ref:`nrf_desktop_hid_forward` |
+|                              |                                  |                   |                                  +--------------------------------+
+|                              |                                  |                   |                                  | :ref:`nrf_desktop_ble_scan`    |
+|                              |                                  |                   |                                  +--------------------------------+
+|                              |                                  |                   |                                  | ``ble_discovery``              |
++------------------------------+----------------------------------+-------------------+----------------------------------+--------------------------------+
```

Comment:
We assume that user will have to read the `Event Manager` documentation (then the user will understand basics of our event-based architecture).

The arrays show the interactions with other modules. They also contain links to the modules to simplify moving around the documentation. I think they are pretty useful.

We were going to generate these tables automatically based on source code (with Python script) - this would simplify the documentation maintenance.

Reply:
I assume you refer to this:
http://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/include/event_manager.html
together with this:
http://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/applications/nrf_desktop/doc/ble_state.html

To be honest, reading this first time didn't help me much, only reading the source code I figured out the intention with the table.
I understand what you try to describe, but the table is much harder to read, compared to a state diagram.